geoportal/geoportailv3_geoportal/static-ngeo/js/catalog/CatalogController.js

```diff
@@ -95,30 +95,36 @@ const exports = function($scope, appThemes, appTheme,
         this.setTree_();
       }), this);
 
-  $scope.$watch(
+  this.scope_.$watch(function() {
```

Comment:
Didn't we try to get rid of $watch by upgrading to > 1.6 ?

Reply:
Good point. Of what I understood, angularjs 1.8 indeed sets the base to prevent `$watch` in cases such as https://github.com/Geoportail-Luxembourg/geoportailv3/pull/2897/commits/76c3c7a89cea2961029ada467014f05e379bb583#diff-a99d939a424ebdf7d29c89240a5127670f51c510d801047de58af923ff4ec0feL830. 
I'm wondering if we can really get rid of all (or most of) the `$watch`(es). As we saw together, it does not seem to make sense here. Thanks for digging into this as well! 
Since they are still used in many places of the application, I propose to look into this for each case when migrating the different parts to web components (step by step).

Reply:
Agreed!